app/src/main/java/com/sunasterisk/boringweather/util/PreferenceUtils.kt

```diff
@@ -0,0 +1,33 @@
+package com.sunasterisk.boringweather.util
+
+import android.content.Context
+import androidx.core.content.edit
+import com.sunasterisk.boringweather.R
+
+object PreferenceUtils {
+    fun getIsFirstLaunch(context: Context, default: Boolean = true) =
+        context.defaultSharedPreferences?.getBoolean(
+            context.getString(R.string.pref_key_is_first_launch),
+            default
+        ) ?: default
+
+    fun setIsFistLaunch(context: Context, value: Boolean) {
+        context.defaultSharedPreferences
+            ?.edit()
+            ?.putBoolean(context.getString(R.string.pref_key_is_first_launch), value)
+            ?.apply()
+    }
+
+    fun getSelectedCityId(context: Context, default: Int = 0) =
+        context.defaultSharedPreferences?.getInt(
+            context.getString(R.string.pref_key_selected_city_id),
+            default
+        ) ?: default
+
+    fun setSelectedCityId(context: Context, value: Int) =
+        context.defaultSharedPreferences?.edit {
+            putInt(context.getString(R.string.pref_key_selected_city_id), value)
+        }
+}
```

Comment:
sao ông không viêt dưới dạng extentions của SharedPreferences nhỉ 
t thấy  context.defaultSharedPreferences lặp lại nhiều

Reply:
Thank you